### PR TITLE
Ports: Add Alpine Mail Client Port

### PR DESCRIPTION
**Pull request by tajmorton**
# Add Alpine Mail Client Port

This PR adds a port of the venerable Alpine mail client (previously Pine).
image

### IMAP over TLS
With an Inbox Path specified in my `.pinerc` as follows:
```
inbox-path={mail.myserver.com/ssl/user=user@myserver.com}Inbox
```
I was able to successfully load my IMAP inbox and view messages:
image
read

### SMTP over TLS
With an SMTP Server setting specified in my `.pinerc` as follows:
```
# List of SMTP servers for sending mail. If blank: Unix Alpine uses sendmail.
smtp-server=mail.myserver.com:465/ssl/user=user@myserver.com

# Sets domain part of From: and local addresses in outgoing mail.
user-domain=myserver.com

# Setup a Pattern Role that sets the From: header to my email address.
# See http://www.ii.com/internet/messaging/pine/changing_from/#whyBest
# for details
patterns-roles=LIT:pattern="/NICK=Default/FLDTYPE=ANY" action="/ROLE=1/FROM=Serenity OS <user@myserver.com>/RTYPE=NC/FTYPE=NC/CTYPE=NC"
```

I was able to successfully compose and send emails from Alpine:
response

### Saving Credentials
To opt into Alpine saving your account passwords, the password file _must already exist_ in the user's directory. I created this with:
```
touch ~/.alpine.pwd
```
Passwords were persisted properly, and the file contained the (obsfucated/encrypted) passwords.
There is a `post_install()` helper to remind people to do this.

## Changes
The first commit is the addition of the port/associated patches to get Alpine compiling and running on Serenity.
Probably the only interesting one here is creating and installing a system-wide configuration file (`/etc/pine.conf`) that sets the default CA certificate file to `/etc/ssl/certs/ca-certificates.ca`.

The second commit is an update to the behavior of `gethostbyname()`/`gethostbyaddr()` that was expose by Alpine. The `hostent*` struct returned by these functions contained a `nullptr` in the `h_aliases` field. The references I found for these functions/this struct all seemed to imply that `h_aliases` would _always_ be an array (possibly with only a `nullptr` element signifying there were no aliases).

I've also included a patch to Alpine so that it is robust to `h_aliases` being `nullptr`, in case that was intentional.

### CODE REVIEW

**Comment by MacDue** [comment]
(No full stops at the end of commit messages)

**Comment by tajmorton** [comment]
> (No full stops at the end of commit messages)

@MacDue thank you for the hint, you are super fast! :)

File: Ports/alpine/package.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=alpine
```
Comment (linusg) [inline Ports/alpine/package.sh:2]: Please single-quote _all_ properties that don't require string interpolation.

File: Ports/alpine/package.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+version=2.26
+useconfigure=true
+configopts=(
+    "--disable-static" \
+    "--enable-shared" \
```
Comment (linusg) [inline Ports/alpine/package.sh:7]: what are the backslashes for?

File: Ports/alpine/package.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+)
+use_fresh_config_sub=true
+use_fresh_config_guess=true
+files="https://alpineapp.email/alpine/release/src/alpine-2.26.tar.xz alpine-${version}.tar.xz c0779c2be6c47d30554854a3e14ef5e36539502b331068851329275898a9baba"
+auth_type=sha256
+depends=("openssl" "ncurses" "ca-certificates")
```
Comment (linusg) [inline Ports/alpine/package.sh:13]: Break into one per line for nicer future diffs

File: Ports/alpine/package.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    run automake --add-missing || true
+    run autoreconf
+}
+
+configure() {
+    run ./"$configscript" \
+        "${configopts[@]}" \
+        --host="${SERENITY_ARCH}-pc-serenity" \
+        --build="$($workdir/config.guess)" \
+        "--with-ssl-lib-dir=${SERENITY_INSTALL_ROOT}/usr/local/lib" \
```
Comment (linusg) [inline Ports/alpine/package.sh:37]: Why are these options quoted before `--foo`?

File: Ports/alpine/patches/0002-Be-robust-to-no-aliases.patch [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/pith/osdep/domnames.c b/pith/osdep/domnames.c
+index c7ea177e65ae08d6163c962934973e966dd54c83..536fa8f3d8dae5d61c306afe49c32233ddc9ddce 100644
+--- a/pith/osdep/domnames.c
++++ b/pith/osdep/domnames.c
+@@ -85,7 +85,7 @@ getdomainnames(char *hostname, int hsize, char *domainname, int dsize)
+ 	 */
+ 	strncpy(hostname, he->h_name, hsize-1);
+ 	hostname[hsize-1] = '\0';
+-	if(strchr(hostname, '.') == NULL){		/* no dot in hostname */
++	if(strchr(hostname, '.') == NULL && he->h_aliases != NULL){		/* no dot in hostname */
```
Comment (linusg) [inline Ports/alpine/patches/0002-Be-robust-to-no-aliases.patch:19]: Why is this patched if you also make a LibC change for it?

**Review by linusg — CHANGES_REQUESTED** [review]

File: Ports/alpine/package.sh (hunk 1 above)
Comment (tajmorton) [reply Ports/alpine/package.sh]: Thank you, done.

File: Ports/alpine/package.sh (hunk 2 above)
Comment (tajmorton) [reply Ports/alpine/package.sh]: Habit, removed.

File: Ports/alpine/package.sh (hunk 3 above)
Comment (tajmorton) [reply Ports/alpine/package.sh]: Good tip, done.

File: Ports/alpine/package.sh (hunk 4 above)
Comment (tajmorton) [reply Ports/alpine/package.sh]: Cleaned this up, only the interpolated parts are quoted now.

File: Ports/alpine/patches/0002-Be-robust-to-no-aliases.patch (hunk 5 above)
Comment (tajmorton) [reply Ports/alpine/patches/0002-Be-robust-to-no-aliases.patch]: Just an artifact of the evolution of this branch, and not being sure if patching libc was the right way to go. This patch has now been dropped. I've built and re-tested.

**Review by linusg — APPROVED** [review]
Exceptionally nice PR for a first-time contributor. Well done!
(also definitely consider upstreaming some of the patches if they are interested)

**Pull request merged**